### DISCUSSION TITLE
Short version of UUID

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#233] [Ideas]
I have a need for a short uuid (to create a unique name for an IndexedDB database) in client code.

Could the UUID module be modified to suit? If so, I would like to have a crack at contributing but will need to be pointed in the right direction.

**Comment by meatballs** [comment]
Not really. UUIDs conform to a known standard, so modifying the existing implementation feels wrong.

You might instead want to have a look at https://hashids.org/ They have a javascript implementation and a wrapper for that would make a nice addition to this library.

**Comment by s-cork** [comment]
You could take a look at https://github.com/skorokithakis/shortuuid/
It's the same package that can be imported in anvil's server modules.

You might be able to adapt some of the functions to work with the `anvil_extras.uuid` module.

Specifically following the rabbit hole starting from `def uuid` method inside this file: https://github.com/skorokithakis/shortuuid/blob/b1b94b8f5e7422d6fc9e15098ab06bdc135560ac/shortuuid/main.py#L80-L97

I'd be ok with `from anvil_extras.uuid import shortuuid`
or maybe `from anvil_extras.shortuuid import uuid`

**Comment by rhurlbatt** [comment]
Understood @meatballs.

Ok great I will go through and see how it looks.

**Comment by rhurlbatt** [comment]
My thought on how to approach was to use a Javascript library as the base so that the short version can be generated on the client - otherwise you would just a python library.

After reading through both links above and doing a bit of searching, I figured that without revinenting the wheel I could use the Short Unique ID library ([https://shortunique.id/](https://shortunique.id/)).

I have done this in my current code in Anvil Native Libraries and it works - maybe not elegantly but it works. Could this be a way forward?

```
<script src="https://cdn.jsdelivr.net/npm/short-unique-id@latest/dist/short-unique-id.min.js"></script>

<script>
  
  function get_short_uuid () {
    
    let uid = new ShortUniqueId({ length: 16});

    var uuidText = uid();

  	return uuidText;
    
  }
</script>
```

**Comment by s-cork** [comment]
I could get behind something like this.
I'd go for a separate module - that way we can take advantage of lazy loading a javascript module without thinking too much.

One issue you'll find @rhurlbatt is including your javascript code inside a python module.
You can't use theme or native libraries which is how you might do this in your own app.

There is a helper function `_add_script()` which you can use.

Other things:
 - use a version rather than `@latest`
 - decide on an api - i.e. what do you want to expose.
   e.g. in your code snippet it would be nice to be able to specify the length somehow
